angular2-jwt.ts

```diff
@@ -15,46 +15,54 @@ export interface IAuthConfig {
   noTokenScheme?:boolean;
 }
 
+export interface IAuthConfigOptional {
+    headerName?: string;
+    headerPrefix?: string;
+    tokenName?: string;
+    tokenGetter?: any;
+    noJwtError?: boolean;
+    globalHeaders?: Array<Object>;
+    noTokenScheme?: boolean;
+}
+
+export class AuthConfigConsts {
+    public static DEFAULT_TOKEN_NAME = 'id_token';
+    public static DEFAULT_HEADER_NAME = 'Authorization';
+    public static HEADER_PREFIX_BEARER = 'Bearer ';
+}
+
+const AuthConfigDefaults: IAuthConfig = {
+    headerName: AuthConfigConsts.DEFAULT_HEADER_NAME,
+    headerPrefix: null,
+    tokenName: AuthConfigConsts.DEFAULT_TOKEN_NAME,
+    tokenGetter: () => localStorage.getItem(AuthConfigDefaults.tokenName),
+    noJwtError: false,
+    globalHeaders: [],
+    noTokenScheme: false
+};
+
 /**
  * Sets up the authentication configuration.
  */
 
 export class AuthConfig {
-  
-  headerName: string;
-  headerPrefix: string;
-  tokenName: string;
-  tokenGetter: any;
-  noJwtError: boolean;
-  noTokenScheme: boolean;
-  globalHeaders: Array<Object>;
 
-  constructor(config:any={}) {
-    this.headerName = config.headerName || 'Authorization';
-    if (config.headerPrefix) {
-      this.headerPrefix = config.headerPrefix + ' ';
-    } else if (config.noTokenScheme) {
-      this.headerPrefix = '';
+  private _config: IAuthConfig;
+
+  constructor(config?: IAuthConfigOptional) {
+    config = config || {};
+    this._config = Object.assign({}, AuthConfigDefaults, config);
```

Comment:
The ponyfill looks pretty good. Can you give it a shot and let me know if it works out?